internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -41,6 +48,9 @@ type HypershiftAgentServiceConfigReconciler struct {
 	// on the management cluster
 	NodeSelector map[string]string
 	Tolerations  []corev1.Toleration
+
+	SpokeK8sClientFactory spoke_k8s_client.SpokeK8sClientFactory
+	SpokeClient           spoke_k8s_client.SpokeK8sClient
```

Comment:
This should not be a field on the reconciler as the same reconciler can handle requests for multiple spokes at the same time.

This would cause some _very_ strange races and bugs.